blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/datastax/DCounterIO.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2016 Rackspace.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * 	http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.rackspacecloud.blueflood.io.datastax;
+
+import com.rackspacecloud.blueflood.io.serializers.metrics.CounterSerDes;
+import com.rackspacecloud.blueflood.types.*;
+
+import java.nio.ByteBuffer;
+
+/**
+ * This class holds the utility methods to read/write counter metrics
+ * using Datastax driver.
+ */
+public class DCounterIO extends DAbstractPreaggregatedIO {
+
+    private CounterSerDes serDes = new CounterSerDes();
+
+    /**
+     * Provides a way for the sub class to get a {@link java.nio.ByteBuffer}
+     * representation of a certain Rollup object.
+     *
+     * @param rollup
+     * @return
+     */
+    @Override
+    protected ByteBuffer toByteBuffer(Rollup rollup) {
+        if ( ! (rollup instanceof BluefloodCounterRollup) ) {
+            throw new IllegalArgumentException("toByteBuffer(): expecting BluefloodCounterRollup class but got " + rollup.getClass().getSimpleName());
+        }
+        BluefloodCounterRollup counterRollup = (BluefloodCounterRollup) rollup;
```

Comment:
Believe me, I tried. I thought I could do this:
```protected ByteBuffer toByteBuffer(<T extends Rollup> rollup)``` in the interface class. But it gave me a compile error when I tried to implement that with the specific Rollup subclass. The error is something like the method is not the same "erasure".

Any suggestions?